e2/src/main/index.ts

```diff
@@ -60,6 +60,107 @@ function createWindow(): void {
       });
     });
 
+    ipcMain.handle('fetchSessionData', async function(_event, partition, url, interactive) {
```

Comment:
Can we do that in backend.ts using JPC, please? It's also running in the main process.

We should not use Electron IPC at all. In fact, avoiding Electron IPC was the whole reason why I invented JPC.

Reply:
Also, please keep the code in the backend to the absolute minimum. Anything OWA-specific should be in app/logic/ (and passed as arguments), not here.

Reply:
I wouldn't be able to make the new window modal to the main window, but apart from that, I didn't realise that those were actually the same process.

Reply:
OK. Non-modal is OK for now.

(Just for your information: e2/src/main/index.ts line 14 calls `startupBackend()`, which is in backend/backend.ts and starts JPC.)

Reply:
It's all OWA-specific because it has to handle the per-session cookies.

Reply:
Per-session cookies useful for many things, e.g. for OAuth2 dialogs.

Reply:
The main purpose is to keep all OWA-specific knowledge in one code place. Esp. "canary" and stuff.

Reply:
I've kept all of the canary specific knowledge here. I didn't think it was a good idea to pass the canary back and forth each time I wanted to make a request...

Reply:
OK, we can disentangle this later. But please do move it into backend and use JPC.
Make a new code file for this, backend/owa.ts .